cura/OAuth2/KeyringAttribute.py

```diff
@@ -39,6 +40,13 @@ def __get__(self, instance: "BaseModel", owner: type) -> Optional[str]:
                 self._store_secure = False
                 Logger.logException("w", "No keyring backend present")
                 return getattr(instance, self._name)
+            except Exception as e:
```

Comment:
Catch-all exceptions are often seen as a code smell. What's wrong with just catching `KeychainDenied` with a specific message, and then catching `KeyringError` with a more general message?

This way, if I make a mistake, e.g. rename `self._keyring_name` without changing the reference to it here, then I'll at least know about it. But problems with getting the password would still be caught without crashing the application.

Reply:
As Jaime noticed, if I try to catch the KeychainDenied in OS's other than OSX, it will crash since it will not be able to find it (KeychainDenied is specific to OSX). So I thought that catching a generic error and then exlpicitly checking for it is the best way.

Reply:
But you could just define a KeychainDenied object in the else statement. As long as it inherits from Exception, that should work.

Reply:
Unfortunately, this breaks mypy (it considers it is a redefinition of KeychainDenied, even though it is conditional).

Reply:
I've made that suggestion as well, to define your own class if not using MacOS. But it is a bit clunky so we decided against that and accepted that we'd be catching all exceptions here then.

I did make a PR for the author to fix this issue: https://github.com/jaraco/keyring/pull/517